### PR TITLE
doc: improve volume cloning doc

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

doc: improve volume cloning doc

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
In order to allow the use of a managed identity, the source storage account requires the Storage Blob Data Reader role, while the destination storage account requires the Storage Blob Data Contributor role. If the Storage Blob Data role is not granted, the CSI driver will use a SAS token as a fallback.

File: deploy/example/cloning/README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ## Prerequisites
 - Make sure that the virtual network, where the driver controller pod is deployed, is added to the allowed list of the storage account's VNet setting.
 - Before proceeding, ensure that the application is not writing data to the source volume.
+- You can assign `Storage Blob Data Contributor` role to controller identity in storage account to use system-wide/user-assigned managed identity to authorize, otherwise CSI will use SAS token. If you clone a volume between different storage accounts, you can just assign `Storage Blob Data Reader` role in source storage account and `Storage Blob Data Contributor` in destination storage account.
```
Comment (andyzhangx) [inline deploy/example/cloning/README.md:10]: In order to allow the use of a managed identity, the source storage account requires the Storage Blob Data Reader role, while the destination storage account requires the Storage Blob Data Contributor role. If the Storage Blob Data role is not granted, the CSI driver will use a SAS token as a fallback.

Comment (umagnus) [reply deploy/example/cloning/README.md]: done

**Review by andyzhangx — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**